lib/ruby_routes/router.rb

```
@@ -1,210 +1,306 @@
         end
 
-        if scope[:defaults]
-          scoped_options[:defaults] = (scoped_options[:defaults] || {}).merge(scope[:defaults])
+        if (defaults = scope[:defaults])
+          scoped_options[:defaults] = (scoped_options[:defaults] || {}).merge(defaults)
         end
       end
 
       scoped_options[:path] = scoped_path
       scoped_options
     end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `apply_scope` is too high. [<13, 24, 9> 28.74/23]

(Metrics/AbcSize)